photon-ml/src/test/scala/com/linkedin/photon/ml/optimization/LogisticRegressionOptimizationProblemTest.scala

```diff
@@ -44,41 +90,118 @@ class LogisticRegressionOptimizationProblemTest {
 
   @Test
   def testInitializeZeroModel(): Unit = {
-    val problem = createProblem
-    val zeroModel = problem.initializeZeroModel(Dimensions)
+    val problem = createProblem()
+    val zeroModel = problem.initializeZeroModel(DIMENSIONS)
 
-    assertEquals(zeroModel.coefficients, Coefficients.initializeZeroCoefficients(Dimensions))
+    assertEquals(zeroModel.coefficients, Coefficients.initializeZeroCoefficients(DIMENSIONS))
   }
 
   @Test
   def testCreateModel(): Unit = {
-    val problem = createProblem
-    val coefficients = generateDenseVector(Dimensions)
+    val problem = createProblem()
+    val coefficients = generateDenseVector(DIMENSIONS)
     val model = problem.createModel(coefficients, None)
 
     assertEquals(model.coefficients.means, coefficients)
   }
 
   @Test
-  def testComputeVariances(): Unit = {
-    val problem = createProblem
+  def testComputeVariancesDisabled(): Unit = {
+    val problem = createProblem()
     val input = mock(classOf[RDD[LabeledPoint]])
-    val coefficients = generateDenseVector(Dimensions)
+    val coefficients = generateDenseVector(DIMENSIONS)
 
-    // TODO: computeVarainces is currently disabled. This test will need to be updated when the default changes
     assertEquals(problem.computeVariances(input, coefficients), None)
   }
+
+  @Test
+  def testComputeVariancesSimple(): Unit = {
+    val problem = createProblem(computeVariance = true)
+    val input = generateUnweightedBenignLocalDataSet
+    val coefficients = generateDenseVector(DIMENSIONS)
+
+    // For logistic regression, the second derivative of the loss function (with regard to z = X_i * B) is:
+    //    sigmoid(z) * (1 - sigmoid(z))
+    val hessianDiagonal: Vector[Double] = input.foldLeft(new DenseVector[Double](DIMENSIONS))
+      { (diagonal: DenseVector[Double], datum: LabeledPoint) =>
+        val features: Vector[Double] = datum.features
+        val z: Double = datum.computeMargin(coefficients)
+        val sigm: Double = sigmoid(z)
+        val d2lossdz2: Double = sigm * (1.0 - sigm)
```

Comment:
Why not use LogisticLossFunction for this computation? For this unit, I think we can assume that it works correctly.